types/share_merging.go

```diff
@@ -212,71 +212,65 @@ func parseMsgShares(shares [][]byte) ([]Message, error) {
 	if len(shares) == 0 {
 		return nil, nil
 	}
-
-	// set the first nid and current share
-	nid := shares[0][:consts.NamespaceSize]
-	currentShare := shares[0][consts.NamespaceSize:]
-	// find and remove the msg len delimiter
-	currentShare, msgLen, err := ParseDelimiter(currentShare)
-	if err != nil {
-		return nil, err
+	// msgs returned
+	msgs := []Message{}
```

Comment:
Named return values are really hard to follow in any non-trivial function. Strongly against them.

Reply:
I'm curious on why? I think named return values increase clarity because they 
1. Remove the need to declare the variable in the function body. This results in one less line of code.
2. Draw attention to the value that will be returned in the function signature. I see this as a readability improvement, especially when the returned variable is modified multiple times in the function body (i.e. an aggregate value like `total` or `msgs` as is the case in this example).

Named return values do enable bare returns which I think are less clear than explicit returns and should be avoided. Fortunately, Go linter [revive](https://golangci-lint.run/usage/linters/#revive) warns on bare returns.

Reply:
It looks like Go lang linter https://github.com/blizzy78/varnamelen has an option to configure named return values but given its not enabled in this repo and not enabled by default.

Also I realize named returns are a subjective code readability thing so I don't want to start a bikeshedding debate on it. Happy to resolve this comment and avoid named returns in this codebase if that's preferred.

Reply:
Bare returns are one big issue. The other is un-obvious side effects. Assigning to a named return value is a side-effect to an otherwise pure function, and there's nothing in the syntax to indicate this explicitly to the reader. Named return values are basically global variables.

Reply:
If we do decide to use them, then we should use them everywhere.

Reply:
Spun out a bikeshed issue: https://github.com/celestiaorg/celestia-core/issues/831. Resolving this comment as a won't do until we reach agreement on a codebase convention.